scripts/obtain/install.sh

```diff
@@ -23,44 +40,76 @@ if [ -t 1 ]; then
     fi
 fi
 
-#Standardise OS name to what is put into filenames of the tarballs.
-current_os()
-{
+say_err() {
+    printf "%b\n" "${red:-}dotnet_install: Error: $1${normal:-}" >&2
+}
+
+say() {
+    # using stream 3 (defined in the beginning) to not interfere with stdout of functions
+    # which may be used as return value
+    printf "%b\n" "${cyan:-}dotnet_install:${normal:-} $1" >&3
+}
+
+say_verbose() {
+    if [ "$verbose" = true ]; then
+        say "$1"
+    fi
+}
+
+get_current_os_name() {
+    eval $invocation
+
     local uname=$(uname)
     if [ "$uname" = "Darwin" ]; then
         echo "osx"
+        return 0
     else
         # Detect Distro
         if [ "$(cat /etc/*-release | grep -cim1 ubuntu)" -eq 1 ]; then
             echo "ubuntu"
+            return 0
         elif [ "$(cat /etc/*-release | grep -cim1 centos)" -eq 1 ]; then
             echo "centos"
+            return 0
         elif [ "$(cat /etc/*-release | grep -cim1 rhel)" -eq 1 ]; then
             echo "rhel"
+            return 0
         elif [ "$(cat /etc/*-release | grep -cim1 debian)" -eq 1 ]; then
             echo "debian"
+            return 0
         fi
     fi
+    
+    say_err "OS name could not be detected"
+    return 1
 }
 
 machine_has() {
-    type "$1" > /dev/null 2>&1
+    eval $invocation
+    
+    which "$1" > /dev/null 2>&1
     return $?
 }
 
-#Not 100% sure at the moment that these checks are enough. We might need to take version into account or do something
-#more complicated. This seemed like a good beginning though as it should catch the default "clean" machine case and give
-#people an appropriate hint.
+check_min_reqs() {
+    if ! machine_has "curl"; then
```

Comment:
In a bunch of scripts in CoreFX and CoreCLR, we also try to use wget if it's present (but curl is not) which is common in some distros.  Can we do that here as well?